### PR TITLE
Log the time it takes to run a getModelExtents query

### PR DESCRIPTION
**Pull request by nick4598**

### CODE REVIEW

File: core/backend/src/CustomViewState3dCreator.ts [hunk 1]
```diff
@@ -61,11 +61,18 @@ export class CustomViewState3dCreator {
     const modelIds = new Set(modelIdsList);
     Logger.logInfo(loggerCategory, "Starting getModelExtents query.");
     for (const id of modelIds) {
+      const modelExtentsStopWatch = new StopWatch("getModelExtents query", false);
       try {
         await new Promise((resolve) => setImmediate(resolve)); // Free up main thread temporarily. Ideally we get queryModelExtents off the main thread and do not need to do this.
+        Logger.logInfo(loggerCategory, "Starting getModelExtents query.");
```
Comment (pmconne) [inline core/backend/src/CustomViewState3dCreator.ts:67]: Shouldn't you log the model Id so we can investigate specific models that take too long?

File: core/backend/src/CustomViewState3dCreator.ts [hunk 2]
```diff
@@ -61,11 +61,18 @@ export class CustomViewState3dCreator {
...
+        modelExtentsStopWatch.start();
         const props = this._imodel.nativeDb.queryModelExtents({ id }).modelExtents;
+        modelExtentsStopWatch.stop();
+        Logger.logInfo(LoggerCategory, `Finished getModelExtents query. Time taken: ${modelExtentsStopWatch.elapsed} ms.`);
```
Comment (wgoehrig) [inline core/backend/src/CustomViewState3dCreator.ts:71]: can we make the elapsed time just log metadata? seems like that will be easier to query than as part of the message...

File: core/backend/src/CustomViewState3dCreator.ts (hunk 1 above)
Comment (nick4598) [reply core/backend/src/CustomViewState3dCreator.ts]: Changed it

File: core/backend/src/CustomViewState3dCreator.ts (hunk 2 above)
Comment (nick4598) [reply core/backend/src/CustomViewState3dCreator.ts]: Changed it

**Review by pmconne — APPROVED** [review]

File: core/backend/src/CustomViewState3dCreator.ts [hunk 3]
```diff
@@ -59,22 +59,29 @@ export class CustomViewState3dCreator {
...
+        Logger.logInfo(loggerCategory, "Starting getModelExtents query.", () => ({modelId: id}));
+        modelExtentsStopWatch.start();
         const props = this._imodel.nativeDb.queryModelExtents({ id }).modelExtents;
+        modelExtentsStopWatch.stop();
+        Logger.logInfo(loggerCategory, "Finished getModelExtents query.", () => ({timeElapsedMs: modelExtentsStopWatch.elapsed, modelId: id}));
         modelExtents.union(Range3d.fromJSON(props), modelExtents);
       } catch (err: any) {
+        modelExtentsStopWatch.stop();
         if ((err as IModelError).errorNumber === IModelStatus.NoGeometry) { // if there was no geometry, just return null range
+          Logger.logInfo(loggerCategory, "Finished getModelExtents query with NoGeometry error.", () => ({timeElapsedMs: modelExtentsStopWatch.elapsed, modelId: id}));
```
Comment (kabentley) [inline core/backend/src/CustomViewState3dCreator.ts:75]: there is no reason to make this an arrow function. Just pass the object.

Comment (nick4598) [reply core/backend/src/CustomViewState3dCreator.ts:75]: Oops, I just copied it from somewhere else in the repo. Changing it now

**Comment by nick4598** [comment]
@mergifyio backport release/3.3.x

**Pull request merged**